### PR TITLE
#2: Java formatter

### PR DESCRIPTION
**Pull request by jkleiber**
**Changes**
- Adds a format checker to CI and as a separate Github Actions workflow
- Configures the [spotless](https://github.com/diffplug/spotless/) plugin to format code on build using the Google style guide

**Decisions**
- Decided to not add a linter at this time, I didn't see the value in adding yet another tool on top of the formatter - we can revisit if it would be helpful

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -38,26 +31,65 @@ public RobotContainer() {
...
-
-        controller.b().onTrue(new InstantCommand(() -> scoringSubsystem.setAction(ScoringSubsystem.ScoringAction.AIM)))
-            .onFalse(new InstantCommand(() -> scoringSubsystem.setAction(ScoringSubsystem.ScoringAction.WAIT)));
-
-        controller.x().onTrue(new InstantCommand(() -> scoringSubsystem.setAction(ScoringSubsystem.ScoringAction.SHOOT)))
-            .onFalse(new InstantCommand(() -> scoringSubsystem.setAction(ScoringSubsystem.ScoringAction.WAIT)));
-
-        controller.y().onTrue(new InstantCommand(() -> scoringSubsystem.setAction(ScoringSubsystem.ScoringAction.ENDGAME)))
-            .onFalse(new InstantCommand(() -> scoringSubsystem.setAction(ScoringSubsystem.ScoringAction.WAIT)));
+        controller
```
Comment (redPlover) [inline src/main/java/frc/robot/RobotContainer.java:34]: Could we change how the formatter handles lambdas?

**Review by redPlover — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (redPlover) [reply src/main/java/frc/robot/RobotContainer.java:34]: Sorry, I meant to highlight 34-84 (controller buttons).

Comment (jkleiber) [reply src/main/java/frc/robot/RobotContainer.java:34]: I'll llok at the settings, but we can probably at least ignore formatting on this block of code so it doesn't look so weird

Comment (jkleiber) [reply src/main/java/frc/robot/RobotContainer.java]: I went with just disabling this section of code formatting for now

**Review by charlesmackenzie — DISMISSED** [review]

**Review by redPlover — APPROVED** [review]
Looks great

**Comment by jkleiber** [comment]
@redPlover @charlesmackenzie had to rebase this due to the other PR going in, should be good for another look

**Review by redPlover — APPROVED** [review]

**Pull request merged**